scripts/dts/edtlib.py

```diff
@@ -169,7 +170,15 @@ def _init_compat2binding(self, bindings_dirs):
             "|".join(re.escape(compat) for compat in dt_compats)
         ).search
 
-        self._binding_paths = _binding_paths(bindings_dirs)
+        self._binding_paths = []
+        # Relative not to leak private and non-deterministic paths
+        relpaths = {}
+
+        for bdir in bindings_dirs:
+            for relp in _binding_rel_yamls(bdir):
+                abs_yaml_path = os.path.join(bdir, relp)
+                self._binding_paths.append(abs_yaml_path)
+                relpaths[abs_yaml_path] = relp
```

Comment:
Tinkered a bit and came up with a `dict`-less version.

```python
        # Used by 'include:'
        self._binding_paths = []
        # Stores paths relative to the binding directory. Used for output, to
        # make it deterministic.
        relpaths = []

        for bdir in bindings_dirs:
            for abspath in _binding_paths(bdir):
                self._binding_paths.append(abspath)
                relpaths.append(os.path.relpath(abspath, bdir))

        self._compat2binding = {}
        for binding_abspath, binding_relpath in zip(self._binding_paths, relpaths):
            with open(binding_abspath, encoding="utf-8") as f:
                contents = f.read()

                # ... (with binding_abspath)

                self._compat2binding[binding_compat, _binding_bus(binding)] = \
                    (binding, binding_relpath)

```

 `zip()` is for iterating both lists at the same time.

`binding_paths()` would be similar to the old version:

```python
def _binding_paths(bindings_dir):
    # Returns a list of paths to all .yaml files in one 'bindings_dir'

    binding_paths = []

    for subdir, _, filenames in os.walk(bindings_dir):
        for filename in filenames:
            if filename.endswith(".yaml"):
                binding_paths.append(os.path.join(subdir, filename))

    return binding_paths
```

Thoughts?

It seems to indirectly remove the `./` at the start of paths too, so would need a test suite update.

Reply:
I'm not sure why this is better than a dict but why not, LGTM.

One of things I liked about the dict is that... it could be done in a small number of changed lines which is easier to rebase constantly. Not the best of reasons, granted :-)